src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -452,7 +458,54 @@ export class SearchInterceptor {
           body: JSON.stringify({
             ...request,
             ...searchOptions,
+            stream:
+              strategy === ESQL_ASYNC_SEARCH_STRATEGY || strategy === ENHANCED_ES_SEARCH_STRATEGY,
           }),
+          asResponse: true,
+        })
+        .then((rawResponse) => {
+          const response = rawResponse.body;
+          const warning = rawResponse.response?.headers.get('warning');
+          const requestParams =
+            (response as Record<string, unknown>).requestParams ??
+            JSON.parse(rawResponse.response?.headers.get('requestParams') || '{}');
+          const error = (response as Record<string, unknown>).error;
```

Comment:
What is this `error` from? Is it only when we're expecting a stream back? I don't see anything in the previous code that checks for this property.

Reply:
yes, we need to manually handle the error when we read the stream